txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+    def get_http_status_code(self) -> int:
+        """Return the HTTP status code for the status."""
+        if self.status == HealthCheckStatus.PASS:
+            return 200
+        elif self.status == HealthCheckStatus.WARN:
+            return 503
+        elif self.status == HealthCheckStatus.FAIL:
+            return 503
+        else:
+            raise ValueError("Invalid status")
+
+    def to_json(self) -> Dict[str, Any]:
+        """Return a dict representation of the object. All field names are converted to camel case."""
+        return {
+            "status": self.status.value,
+            "description": self.description,
+            "checks": {k: [c.to_json() for c in v] for k, v in self.checks.items()},
+        }
+
+
+class ComponentHealthCheckInterface(ABC):
+    """This is an interface to be used by other classes implementing health checks for components."""
+
+    @abstractmethod
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the health check status for the component."""
+        raise NotImplementedError()
+
+
+class HealthCheck:
+    """This is the main class that will use the other classes to check the health of the components.
+
+    It will aggregate the responses into a final object to be returned following our standards.
+    """
+
+    def __init__(self, manager: "TxMiningManager", backend: "HathorClient") -> None:
+        """Init the class with the components that will be checked."""
+        self.health_check_components: List[ComponentHealthCheckInterface] = [
+            MiningHealthCheck(manager),
+            FullnodeHealthCheck(backend),
+        ]
+
+    async def get_health_check(self) -> HealthCheckResult:
+        """Return the health check status for the tx-mining-service."""
+        components_health_checks = [
+            await c.get_health_check() for c in self.health_check_components
+        ]
+        components_status = {c.status for c in components_health_checks}
+
+        overall_status = HealthCheckStatus.PASS
+        if HealthCheckStatus.FAIL in components_status:
+            overall_status = HealthCheckStatus.FAIL
+        elif HealthCheckStatus.WARN in components_status:
+            overall_status = HealthCheckStatus.WARN
+
+        return HealthCheckResult(
+            status=overall_status,
+            description="health of tx-mining-service",
+            checks={c.component_name: [c] for c in components_health_checks},
+        )
+
+
+class FullnodeHealthCheck(ComponentHealthCheckInterface):
+    """This class will check the health of the fullnode by sending a request to its /v1a/health."""
+
+    def __init__(self, backend: "HathorClient") -> None:
+        """Init the class with the fullnode backend."""
+        self.backend = backend
+
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the fullnode health check status."""
+        health_check = ComponentHealthCheck(
+            component_name="fullnode",
```

Comment:
Maybe move component names to a constant, or even better an enum?

Reply:
Done in https://github.com/HathorNetwork/tx-mining-service/pull/104/commits/acc340e690cfd4bf508fa18b6b25a2a98385b5aa